Wizardry/Core/StatScreen/DrawPages/DrawPage1.c

```diff
@@ -231,51 +350,65 @@ STATIC_DECLAR void DrawPage1ValueReal(void)
 {
     struct Unit * unit = gStatScreen.unit;
 
-    DrawStatWithBarRework(0, 0x5, 0x1,
-                    unit->pow,
-                    GetUnitPower(unit),
-                    UNIT_POW_MAX(unit));
-
-    DrawStatWithBarRework(1, 0x5, 0x3,
-                    UNIT_MAG(unit),
-                    GetUnitMagic(unit),
-                    GetUnitMaxMagic(unit));
-
-    DrawStatWithBarRework(2, 0x5, 0x5,
-                    unit->skl,
-                    GetUnitSkill(unit),
-                    UNIT_SKL_MAX(unit));
-
-    DrawStatWithBarRework(3, 0x5, 0x7,
-                    unit->spd,
-                    GetUnitSpeed(unit),
-                    UNIT_SPD_MAX(unit));
-
-    DrawStatWithBarRework(4, 0x5, 0x9,
-                    unit->lck,
-                    GetUnitLuck(unit),
-                    UNIT_LCK_MAX(unit));
-
-    DrawStatWithBarRework(5, 0x5, 0xB,
-                    unit->def,
-                    GetUnitDefense(unit),
-                    UNIT_DEF_MAX(unit));
-
-    DrawStatWithBarRework(6, 0x5, 0xD,
-                    unit->res,
-                    GetUnitResistance(unit),
-                    UNIT_RES_MAX(unit));
+    if (DisplayGrowths()) 
+    { 
+        DrawGrowthWithDifference(0x4, 0x1, GetUnitBasePowGrowth(unit), GetUnitPowGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0x3, GetUnitBaseMagGrowth(unit), GetUnitMagGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0x5, GetUnitBaseSklGrowth(unit), GetUnitSklGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0x7, GetUnitBaseSpdGrowth(unit), GetUnitSpdGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0x9, GetUnitBaseLckGrowth(unit), GetUnitLckGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0xB, GetUnitBaseDefGrowth(unit), GetUnitDefGrowth(unit));
+        DrawGrowthWithDifference(0x4, 0xD, GetUnitBaseResGrowth(unit), GetUnitResGrowth(unit));
+        DrawGrowthWithDifference(0xC, 0x1, GetUnitBaseHpGrowth(unit),  GetUnitHpGrowth(unit));
```

Comment:
you can directly return at here so that the rest of the function can escape from modification